pom.xml

```diff
@@ -61,7 +61,7 @@
       <groupId>com.github.spotbugs</groupId>
       <artifactId>spotbugs-annotations</artifactId>
       <version>4.0.1</version>
-      <scope>provided</scope>
+      <scope>test</scope>
```

Comment:
spotbugs annotations are required in runtime code. 
What we had here is equivalent to EE, where spotbugs is put globally in all sub modules in provided scope by the build.gradle file.

If we do not want to enforce maven projects to use spotbugs, then we can remove the dependency. But that would be sad. 

core is an exception... I am not sure this is right to use an exception as a global rule and consequently remove spotbug.

It is like in EE we would remove spotbug from the global gradle file because we have one sub project below that is not using it...

Reply:
It is required for scanning runtime code in the verify lifecycle, not used at runtime.  I tested this change tc-platform with `mvnw clean install` and everything worked fine.  This change does not require any changes to any downstream projects.

Reply:
So the dependency is not required to compile the main source after all?

Reply:
looks like i'm wrong about this.  My build was pulling a stale version of snapshot.  I'll have to find a different way.

Reply:
This should be satisfactory to all consumers now.

Reply:
If move to `test` scope:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project common-structures: Compilation failure: Compilation failure: 
[ERROR] /Users/matc/workspace/terracotta/terracotta-platform/common/structures/src/main/java/org/terracotta/common/struct/MemoryUnit.java:[18,38] error: package edu.umd.cs.findbugs.annotations does not exist
[ERROR] /Users/matc/workspace/terracotta/terracotta-platform/common/structures/src/main/java/org/terracotta/common/struct/MemoryUnit.java:[88,3] error: cannot find symbol
[ERROR]   symbol:   class SuppressFBWarnings
[ERROR]   location: class MemoryUnit
```
Because we need this annotations when compiling runtime code.

Reply:
See my latest update.  I moved it to a profile that can be disabled.  That should work for you.